api/core/v1alpha1/core.proto

```diff
@@ -0,0 +1,42 @@
+syntax = "proto3";
+
+package core.skv2.solo.io;
+
+option go_package = "github.com/solo-io/skv2/pkg/api/core.skv2.solo.io/v1alpha1/types";
```

Comment:
not sure i agree with the choice of v1alpha1 for the version for core types. 

also, what's the reasoning using a `types` package rather than a package named for the api version?

Reply:
Why no v1alpha1, using the versioning just allows us to make breaking changes to the API without having to make breaking changes to the code. As we support more APIs, and potentially multiple versions of those APIs, following kubernetes versioning for our APIs will become more and more useful.